src/scatter.jl

```diff
@@ -148,10 +148,22 @@ end
 function ∇scatter_src(::typeof(mean), Δ, dst,
                     src::AbstractArray{Tsrc,Nsrc}, 
                     idx::AbstractArray{Tidx,Nidx}) where {Tsrc,Tidx,Nsrc,Nidx}
-    dims = Nsrc - Nidx
-    divide_by_counts!(gather(Δ, idx), idx, dims)
+    
+    M = typelength(Tidx)
+    num = gather(Δ, idx)
+    print("START")
+    counts = fill!(similar(Δ, Int, size(Δ)[end-M+1:end]), 0)
+    scatter!(+, counts, fill!(similar(idx), 1), idx)
+    den = gather(counts, idx)
+    
+    # make num and den broadcast compatible
+    for i in 1:ndims(num)-ndims(den)
+        den = unsqueeze(den)
+    end
+    return safe_div.(num, den)
```

Comment:
`save_div` looks odd, is there precedent to doing this sort of division?

Reply:
it's already used in the scatter-mean forward pass. Here it's needed to correctly backpropagate zeros instead of infs

Reply:
Could you link to where pytorch/ tf does this?

Reply:
I'm not familiar with those codebases

Reply:
Just as a side note: the most relevant pytorch code is in these directories:

https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native
https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/cpu
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cpu/ScatterGatherKernel.cpp